### PR TITLE
fix(dependabot): three classifier correctness fixes (auth-critical patch, multi-package titles, stylelint tooling)

### PR DESCRIPTION
**Pull request by joshsmithxrm**
Three dependabot-classifier correctness fixes, all surfaced by the same `/dependabot-triage` run. `docs/MERGE-POLICY.md` is authoritative and unchanged — these are code-catches-up-to-doc fixes.

---

## Fix 1 — auth-critical PATCH bumps wrongly auto-merge-eligible (real PR #1288)

**Policy:** auth-critical packages (`Microsoft.Identity.Client`, `Azure.Identity`, `System.Security.Cryptography.Pkcs`, …) go through **Group B (verify-then-merge)** at **any non-major version** — patch included.

**Bug:** the single-PR path in `classify_pr()` gated the override on `update_type == "minor"` only, so an auth-critical **patch** bump fell through to the generic `patch → Group A` branch (auto-merge). Caught on PR #1288 (`System.Security.Cryptography.Pkcs` 10.0.5→10.0.9).

**Fix:** `is_auth_critical_package(pkg) and update_type in ("patch", "minor")`, and the reason string now interpolates the actual `update_type`. `major` already returns Group C earlier; auth-critical `unknown` still falls through to the Group-C manual-review default. The grouped-PR path was already correct and is untouched.

## Fix 2 — multi-package "Bump X and Y" titles not parsed (real PR #1299)

**Bug:** a title like `Bump Microsoft.Identity.Client and Microsoft.Identity.Client.Extensions.Msal` carries no version, so `parse_title()` returned `(None, None, None)`, `update_type` became `"unknown"`, and `classify_pr()` fell through to the Group-C "could not classify" default — even though the body enumerates both members as auth-critical **minor** bumps that should be **Group B**.

**Fix:** when the single-title parse yields no package and the title is not a `group:` title, attempt `parse_group_members(body)`; if it finds ≥1 members, route them through the **same** per-member most-conservative-wins logic the grouped path uses. That block was factored out of the grouped branch into a shared `_classify_member_set()` helper, so the `group:` path and the new multi-package path reuse it (no duplication). A genuinely unparseable PR (no title parse **and** no body members) still hits the Group-C default. Normal single-package and `group:` PRs are unchanged.

## Fix 3 — `stylelint` missing from `TOOLING_PACKAGES` (real PR #1279)

**Bug:** `stylelint` is a dev-time CSS linter (same class as eslint/knip/prettier), but its absence routed a stylelint minor bump to Group B instead of Group A.

**Fix:** added `"stylelint"` to the `TOOLING_PACKAGES` frozenset.

---

## Tests

Added regression coverage in `tests/scripts/dependabot/test_classify.py`:
- auth-critical **PATCH** (`Pkcs` 10.0.5→10.0.9) → **B**; auth-critical **MAJOR** (`Identity.Client` 4→5) still → **C**
- multi-package auth-critical minor (#1299) → **B** (reason names the auth-critical package); multi-package with a major member → **C**; two non-critical patch members → **A**; unparseable body → **C** "could not classify"
- `stylelint` minor (#1279) → **A** (tooling)

Existing auth-critical minor → B, non-auth-critical patch → A, all `group:` cases, and `TestAuthCriticalDocCodeDrift` remain green (no change touches the auth-critical list). Full suite: **64 passed**.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved dependency update classification for multi-package PRs by evaluating each member and applying the most-conservative A/B/C outcome, with consistent safety guards.
  * Enhanced auth-critical handling: auth-critical patch/minor bumps now map to Group B (major remains Group C) with bump-specific reasons.
  * Expanded tooling attribution allowlist to include `stylelint`.
* **Tests**
  * Added regression coverage for tooling-style packages, auth-critical patch/minor/major, multi-package edge cases (including BREAKING CHANGE forcing Group C), and reason-message assertions.

### CODE REVIEW

**Comment by joshsmithxrm** [comment]
@coderabbitai review

**Comment by joshsmithxrm** [comment]
@coderabbitai review

**Comment by joshsmithxrm** [comment]
C:/Program Files/Git/gemini review

File: scripts/dependabot/classify.py [hunk 1]
```diff
@@ -391,54 +458,24 @@ def classify_pr(pr: dict) -> Classification:
...
+    # logic as grouped PRs so auth-critical members aren't lost to the Group-C
+    # "could not classify" default (real case: PR #1299 bumping two auth-critical
+    # packages). A genuinely unparseable PR (no members) falls through to that
+    # default below.
+    if pkg is None:
+        members = parse_group_members(body)
+        if members:
+            return _classify_member_set(
+                number, None, "multi-package bump", members, ecosystem
             )
```
[the remark this answers is not in the document]
Comment (joshsmithxrm) [reply scripts/dependabot/classify.py:502]: Addressed in 180e6369b. Threaded `body` and `files` into `_classify_member_set()` and applied both guards as most-conservative overrides before it returns: `changelog_signals_breaking(body)` now forces Group C, and `touches_auth_critical_path(files)` escalates to at least Group B — matching the single-package path. Added regression tests for grouped + multi-package BREAKING CHANGE markers and member-set auth-critical-path escalation. `pytest tests/scripts/dependabot/` → 67 passed.

@coderabbitai resolve
